include/core/mir/mutex.h

```diff
@@ -92,6 +113,15 @@ class Mutex
         return MutexGuard<Guarded>{std::unique_lock{mutex}, value};
     }
 
+    /**
+     * Unlock an acquired mutex
+     *
+     * This is a convenience method for cases when the mutex must be dropped
+     * but cannot easily be made to leave scope.
+     */
+    static void drop(MutexGuard<Guarded> /*to_drop*/)
+    {
+    }
 private:
     std::mutex mutex;
```

Comment:
It is usual for a `mutex` to be `mutable`, the only reason that isn't needed here is there's been attempt at const-correctness which means that mutability has been propagated to the instance.

On the surface, it would appear to be possible to provide const overloads that return const `MutexGuard<Guarded const>` and so maintain const correctness.